### ISSUE TITLE
Undefined method Connection::getEventManager

### ISSUE THREAD

**Issue by mihai-stancu** [phpstan/phpstan-doctrine#661]
Hello,

I've identified an issue with this plugin that started appearing recently. I'm trying to isolate it into small repro steps.

#### Error:
In our CI the following error is reported:
```
Internal error: Call to undefined method
Doctrine\DBAL\Connection::getEventManager() while analysing file: <not always the same file>

Run PHPStan with -v option and post the stack trace to:                                             
https://github.com/phpstan/phpstan/issues/new?template=Bug_report.yaml
```

Locally (inside our Docker image) we get one of a few outcomes:
1) The above error -- the file it is reported at varies.
2) Process hangs at 80/534
3) Process hangs at 100/534
4) Process hangs at 120/534

#### Chronology:
We build our pipeline images weekly so perhaps this is caused by a recent release either from PHPStan or phpstan-doctrine from within the last ~10 days.

#### Steps to isolate:
1) Ran PHPStan with all plugins except for phpstan-doctrine plugin -- scans worked (without internal errors)
2) Ran PHPStan with only the phpstan-doctrine plugin no other configuration files -- same error / hanging behavior

#### Versions:
> PHPStan - PHP Static Analysis Tool 2.1.16
Is installed in a **separate** directory from the application `/opt/phpstan/vendor/phpstan/phpstan/`.

> phpstan/phpstan-doctrine version: 2.0.3
Is installed in a **separate** directory from the application `/opt/phpstan/vendor/phpstan/phpstan-doctrine/`.

> doctrine/dbal version: **3.9.4**
The DBAL is a project requirement so it's installed in `/app/vendor/doctrine/dbal`

#### Stack trace:
```
Internal error: Call to undefined method Doctrine\DBAL\Connection::getEventManager() while analysing file /app/src/<some project file>
Post the following stack trace to https://github.com/phpstan/phpstan/issues/new?template=Bug_report.yaml:                                                                                                 
## /app/vendor/doctrine/orm/src/EntityManager.php(166)                                                                                                                                                    
#0 /app/vendor/doctrine/orm/src/EntityManager.php(1006): Doctrine\ORM\EntityManager->__construct()                                                                                                        
#1 /opt/phpstan/vendor/phpstan/phpstan-doctrine/src/Doctrine/Mapping/ClassMetadataFactory.php(53): Doctrine\ORM\EntityManager::create()                                                                   
#2 /app/vendor/doctrine/persistence/src/Persistence/Mapping/AbstractClassMetadataFactory.php(411): PHPStan\Doctrine\Mapping\ClassMetadataFactory->initialize()                                            
#3 /opt/phpstan/vendor/phpstan/phpstan-doctrine/src/Type/Doctrine/ObjectMetadataResolver.php(83): Doctrine\Persistence\Mapping\AbstractClassMetadataFactory->isTransient()                                
#4 /opt/phpstan/vendor/phpstan/phpstan-doctrine/src/Type/Doctrine/ObjectMetadataResolver.php(114): PHPStan\Type\Doctrine\ObjectMetadataResolver->isTransient()                                            
#5 /opt/phpstan/vendor/phpstan/phpstan-doctrine/src/Rules/Doctrine/ORM/PropertiesExtension.php(68): PHPStan\Type\Doctrine\ObjectMetadataResolver->getClassMetadata()                                      
#6 phar:///opt/phpstan/vendor/phpstan/phpstan/phpstan.phar/src/Node/ClassPropertiesNode.php(148): PHPStan\Rules\Doctrine\ORM\PropertiesExtension->isInitialized()                                         
#7 phar:///opt/phpstan/vendor/phpstan/phpstan/phpstan.phar/src/Rules/DeadCode/UnusedPrivatePropertyRule.php(178): PHPStan\Node\ClassPropertiesNode->getUninitializedProperties()                          
#8 phar:///opt/phpstan/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/FileAnalyser.php(108): PHPStan\Rules\DeadCode\UnusedPrivatePropertyRule->processNode()                                            
#9 phar:///opt/phpstan/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php(842): PHPStan\Analyser\FileAnalyser->PHPStan\Analyser\{closure}()                                           
#10 phar:///opt/phpstan/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php(468): PHPStan\Analyser\NodeScopeResolver->processStmtNode()                                                
#11 phar:///opt/phpstan/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php(792): PHPStan\Analyser\NodeScopeResolver->processStmtNodes()                                               
#12 phar:///opt/phpstan/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php(416): PHPStan\Analyser\NodeScopeResolver->processStmtNode()                                                
#13 phar:///opt/phpstan/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/FileAnalyser.php(175): PHPStan\Analyser\NodeScopeResolver->processNodes()                                                        
#14 phar:///opt/phpstan/vendor/phpstan/phpstan/phpstan.phar/src/Command/WorkerCommand.php(136): PHPStan\Analyser\FileAnalyser->analyseFile()                                                              
#15 phar:///opt/phpstan/vendor/phpstan/phpstan/phpstan.phar/vendor/evenement/evenement/src/EventEmitterTrait.php(111): PHPStan\Command\WorkerCommand::PHPStan\Command\{closure}()                         
#16 phar:///opt/phpstan/vendor/phpstan/phpstan/phpstan.phar/vendor/clue/ndjson-react/src/Decoder.php(117): _PHPStan_f61439f6a\Evenement\EventEmitter->emit()                                              
#17 phar:///opt/phpstan/vendor/phpstan/phpstan/phpstan.phar/vendor/evenement/evenement/src/EventEmitterTrait.php(111): _PHPStan_f61439f6a\Clue\React\NDJson\Decoder->handleData()                         
#18 phar:///opt/phpstan/vendor/phpstan/phpstan/phpstan.phar/vendor/react/stream/src/Util.php(62): _PHPStan_f61439f6a\Evenement\EventEmitter->emit()                                                       
#19 phar:///opt/phpstan/vendor/phpstan/phpstan/phpstan.phar/vendor/evenement/evenement/src/EventEmitterTrait.php(111): _PHPStan_f61439f6a\React\Stream\Util::_PHPStan_f61439f6a\React\Stream\{closure}()  
#20 phar:///opt/phpstan/vendor/phpstan/phpstan/phpstan.phar/vendor/react/stream/src/DuplexResourceStream.php(168): _PHPStan_f61439f6a\Evenement\EventEmitter->emit()                                      
#21 phar:///opt/phpstan/vendor/phpstan/phpstan/phpstan.phar/vendor/react/event-loop/src/StreamSelectLoop.php(201): _PHPStan_f61439f6a\React\Stream\DuplexResourceStream->handleData()                     
#22 phar:///opt/phpstan/vendor/phpstan/phpstan/phpstan.phar/vendor/react/event-loop/src/StreamSelectLoop.php(173): _PHPStan_f61439f6a\React\EventLoop\StreamSelectLoop->waitForStreamActivity()           
#23 phar:///opt/phpstan/vendor/phpstan/phpstan/phpstan.phar/src/Command/WorkerCommand.php(96): _PHPStan_f61439f6a\React\EventLoop\StreamSelectLoop->run()                                                 
#24 phar:///opt/phpstan/vendor/phpstan/phpstan/phpstan.phar/vendor/symfony/console/Command/Command.php(259): PHPStan\Command\WorkerCommand->execute()                                                     
#25 phar:///opt/phpstan/vendor/phpstan/phpstan/phpstan.phar/vendor/symfony/console/Application.php(870): _PHPStan_f61439f6a\Symfony\Component\Console\Command\Command->run()                              
#26 phar:///opt/phpstan/vendor/phpstan/phpstan/phpstan.phar/vendor/symfony/console/Application.php(261): _PHPStan_f61439f6a\Symfony\Component\Console\Application->doRunCommand()                         
#27 phar:///opt/phpstan/vendor/phpstan/phpstan/phpstan.phar/vendor/symfony/console/Application.php(157): _PHPStan_f61439f6a\Symfony\Component\Console\Application->doRun()                                
#28 phar:///opt/phpstan/vendor/phpstan/phpstan/phpstan.phar/bin/phpstan(114): _PHPStan_f61439f6a\Symfony\Component\Console\Application->run()                                                             
#29 phar:///opt/phpstan/vendor/phpstan/phpstan/phpstan.phar/bin/phpstan(115): _PHPStan_f61439f6a\{closure}()                                                                                              
#30 /opt/phpstan/vendor/phpstan/phpstan/phpstan(8): require('...')                                                                                                                                        
#31 /usr/local/bin/phpstan(119): include('...')                                                                                                                                                           
#32 {main} 
```

**Comment by staabm** [comment]
`Doctrine\DBAL\Connection::getEventManager` only exists in doctrine/dbal 3.x
Are you sure your project is not runnung on doctrine/dbal 4.x which dropped this method..

**Comment by mihai-stancu** [comment]
I my project's dependencies (`/app/vendor`) I have v3.9.4.

In PHPStan's dependencies (`/opt/phpstan/vendor`) I have v4.2.3.

I'll test with PHPStan + PHPStan Doctrine installed in the project which will likely work.

**Comment by ondrejmirtes** [comment]
Install PHPStan in your project's composer.json.

**Comment by mihai-stancu** [comment]
Thank you for your time.
